atomic_reactor/plugins/post_push_operator_manifest.py

```diff
@@ -0,0 +1,179 @@
+            self.log.info('Isolated build. Skipping')
+            return False
+
+        return True
+
+    def get_koji_operator_manifest_url(self):
+        """Construct URL for downloading manifest from koji task work server_dir
+
+        :rtype: str
+        :return: URL of koji operator manifests archive
+        """
+
+        server_dir = get_koji_upload_dir(self.workflow)
+
+        pathinfo = get_koji_path_info(self.workflow, NO_FALLBACK)
+        file_url = "{}/work/{}/{}".format(
+            pathinfo.topdir, server_dir, OPERATOR_MANIFESTS_ARCHIVE)
+
+        return file_url
+
+    def download_file(self, url, dest_filename, insecure=False):
+        """Downloads file specified by URL
+
+        :param url: file url
+        :param dest_filename: filename to be used for downloaded content
+        :param insecure: download file without cert validation
+
+        :return: file path of downloaded content
+        """
+        self.log.debug('Downloading file: %s', url)
+
+        workdir = self.workflow.source.get_build_file_path()[1]
+        dest_dir = os.path.join(workdir, self.DOWNLOAD_DIR)
+        dest_path = os.path.join(dest_dir, dest_filename)
+        if not os.path.exists(dest_dir):
+            os.makedirs(dest_dir)
+
+        req_session = get_retrying_requests_session()
+        request = req_session.get(url, stream=True,
+                                  verify=not insecure)
+        request.raise_for_status()
+
+        with open(dest_path, 'wb') as f:
+            for chunk in request.iter_content(chunk_size=DEFAULT_DOWNLOAD_BLOCK_SIZE):
+                f.write(chunk)
+
+        self.log.debug('Download finished: %s', dest_path)
+
+        return dest_path
+
+    def run(self):
+        """
+        Run the plugin.
+
+        :return: dictionary with following keys:
+          - endpoint: appregistry API endpoint
+          - registryNamespace: namespace/organization where operator manifest was pushed
+          - repository: repository name
+          - release: release version (related to appregistry repository)
+        """
+
+        if not self.should_run():
+            return
+
+        omps_config = get_omps_config(self.workflow)
+        omps = OMPS.from_config(omps_config)
+
+        operator_manifest_url = self.get_koji_operator_manifest_url()
+        operator_manifests_path = self.download_file(
+            operator_manifest_url, OPERATOR_MANIFESTS_ARCHIVE,
+            insecure=omps_config.get('koji_insecure', False)
+        )
+
+        try:
+            with open(operator_manifests_path, 'rb') as fb:
+                result = omps.push_archive(fb)
+        except OMPSError as e:
+            msg = "Failed to push operator manifests: {}".format(e)
+            self.log.error(msg)
+            raise RuntimeError(msg)
+
+        try:
+            os.remove(operator_manifests_path)
+        except OSError as e:
+            self.log.warning("Cleanup of the downloaded archive failed: %s", e)
```

Comment:
This is missing coverage from the unit tests. Could we have a test case for this one?

Reply:
Do we really need test for this?